x-pack/plugins/security_solution/public/common/components/exceptions/viewer/exceptions_viewer_header.tsx

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  EuiFieldSearch,
+  EuiButtonGroup,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPopover,
+  EuiContextMenu,
+  EuiButton,
+  EuiContextMenuPanelDescriptor,
+  EuiButtonGroupOption,
+} from '@elastic/eui';
+import React, { useState, useCallback, useMemo } from 'react';
+
+import * as i18n from '../translations';
+import { ToggleId } from '../types';
+
+interface ExceptionsViewerHeaderProps {
+  selectedListType: ToggleId;
+  isInitLoading: boolean;
+  listTypes: ToggleId[];
+  onFiltersChange: ({ filter, tags }: { filter: string; tags: string[] }) => void;
+  onAddExceptionClick: (type: ToggleId) => void;
+  onToggleListType: (type: ToggleId) => void;
+}
+
+/**
+ * Collection of filters and toggles for filtering exception items.
+ */
+const ExceptionsViewerHeaderComponent = ({
+  selectedListType,
+  isInitLoading,
+  listTypes,
+  onFiltersChange,
+  onToggleListType,
+  onAddExceptionClick,
+}: ExceptionsViewerHeaderProps): JSX.Element => {
+  const [isAddExceptionMenuOpen, setAddExceptionMenuOpen] = useState(false);
+
+  const onToggle = useCallback(
+    (id: string): void => {
+      const toggle =
+        id === ToggleId.DETECTION_ENGINE ? ToggleId.DETECTION_ENGINE : ToggleId.ENDPOINT;
+      onToggleListType(toggle);
+    },
+    [onToggleListType]
+  );
+
+  const handleOnSearch = useCallback(
```

Comment:
I noticed that in other places like the Rules list page and the Cases list page, we don't make the call to the API until the user presses "Enter". Do we want to do the same here? If not, would it be a good idea to debounce the calls?

Reply:
Yeah, we probably want to go that route. While I think the ideal UX would be search on type (debounced) with a small loading spinner right-aligned in the search box to show activity, we currently only do that when searching in-memory table-data (e.g. the ML Job Settings table). For consistency and ease I think it's fine to search on `Enter`, and show the main loader like we do with Rules/Cases.

Reply:
I've got a follow up PR lined up with this change.